tests/Shared/SharedArchitectureTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodelyTv\Tests\Shared;
+
+use ArrayIterator;
+use BackedEnum;
+use CodelyTv\Backoffice\Auth\Application\Authenticate\AuthenticateUserCommand;
+use CodelyTv\Tests\Shared\Infrastructure\Doctrine\MySqlDatabaseCleaner;
+use Countable;
+use DateTimeImmutable;
+use DateTimeInterface;
+use DomainException;
+use InvalidArgumentException;
+use IteratorAggregate;
+use PHPat\Selector\Selector;
+use PHPat\Test\Builder\Rule;
+use PHPat\Test\PHPat;
+use Ramsey\Uuid\Uuid;
+use ReflectionClass;
+use RuntimeException;
+use Stringable;
+use Throwable;
+use Traversable;
+
+final class SharedArchitectureTest
+{
+    public function test_shared_domain_should_not_import_from_outside(): Rule
+    {
+        return PHPat::rule()
+            ->classes(Selector::inNamespace('CodelyTv\Shared\Domain'))
+            ->canOnlyDependOn()
+            ->classes(...array_merge($this->languageClasses(), [
```

Comment:
Maybe you want to set `ignore_built_in_classes: true` in the configuration
https://www.phpat.dev/documentation/configuration/

(`true` will be the default value in the next minor version `v0.11.0`)

Reply:
For now, I prefer to keep it disabled, as there are some built-in classes that we consider to be infrastructure (such as Reflection), and we might overlook them.

I would rather manually add all domain built-in classes, accepting the possibility of forgetting to add an Infrastructure class and not receiving an error. 🙌